lib/vty.c

```diff
@@ -2651,25 +2637,20 @@ static void vty_event(enum event event, int sock, struct vty *vty)
 		vector_set_index(Vvty_serv_thread, sock, vty_serv_thread);
 		break;
 	case VTYSH_READ:
-		vty->t_read = NULL;
```

Comment:
If you remove this `t_*` `NULL` attributions the next schedule it won't work!

Check out the `thread_add_(read|write)` code:

```c
struct thread *funcname_thread_add_read_write(int dir, struct thread_master *m,
					      int (*func)(struct thread *),
					      void *arg, int fd,
					      struct thread **t_ptr,
					      debugargdef)
{
...
		if (t_ptr
		    && *t_ptr) // thread is already scheduled; don't reschedule
		{
			pthread_mutex_unlock(&m->mtx);
			return NULL;
		}
...
}
```

I think you have to decide where and when you are going to set these variables to `NULL` in this code. I would set them on their callbacks instead of here, but you've also removed them there!

Reply:
@rzalamena -> I think you are missunderstanding how thread code works.